src/fu-util.c

```diff
@@ -2481,6 +2481,15 @@ main (int argc, char *argv[])
 			    "is no longer supported by the upstream developers!\n");
 	}
 
+#ifdef HAVE_SYSTEMD
+	/* make sure the correct daemon is in use */
+	if (!fu_util_using_correct_daemon (&error)) {
+		g_printerr ("Daemon verification failed: %s\n",
```

Comment:
What should the user do if this error is printed? Is the user expected to always leave one single fwupd in the system (packaged or snap) and having two is an error?

Reply:
If you look at https://github.com/hughsie/fwupd/wiki/fwupd-snap you'll see a mention that it's recommended to not use them at the same time.

_Generally_ there isn't a big conflict with leaving both on the system, but it can be an unexpected experience where certain features don't operate properly.

* As an example, if a device flag was introduced in the daemon and you're running the snapped daemon but using a hand install or packaged install client, you might see an `(unknown)` in `Flags` when you call `fwupdmgr get-devices`.

* Or if you are using the reverse then it's possible that a client feature won't work when you talk to the daemon if it's an older daemon.

The reason that this all occurs is that both install `org.freedesktop.fwupd.service` which controls which systemd unit/daemon will start when dbus activation is used by a frontend.

> What should the user do if this error is printed?

I'm not sure yet.  That's up for us to decide what to do.  We could downgrade this to a `WARNING` and tell the user that undefined behavior may occur mixing and matching snap/packaged versions.

Another alternative to this might be for the client to examine the daemon version and if the daemon isn't client or newer show a warning.

Reply:
> Another alternative to this might be for the client to examine the daemon version and if the daemon isn't client or newer show a warning.

Pushed a commit that does just that.  So we can pick and choose what makes most sense here.

Reply:
> If you look at https://github.com/hughsie/fwupd/wiki/fwupd-snap you'll see a mention that it's recommended to not use them at the same time.

I believe the source of the error was that the user had a system-installed fwupd via my PPA, and then they tried to also test the standalone installer script on the same platform, without using the explicit "--uninstall" to get the snap removed after the standalone installer has finished. That --uninstall when calling the installer is a bit misleading BTW, maybe something like "--cleanup-after install" would be clearer?

Reply:
If it's from the installer script that's actually good - we can probably control that experience better to guide them to removing the fwupd deb and installing the snap that is included in it. Another alternative can be to hard code in the installer script some sort of "minimum version" and give them an option to use system deb if they have a new enough daemon installed already.